README.md

```diff
@@ -11,78 +11,87 @@
 of connectivity between the cluster's nodes. Network checker consists of two
 parts: server (this repository) and agent
 ([developed here](https://github.com/Mirantis/k8s-netchecker-agent)). Agents
-are deployed on every K8S node using
-[Daemonset mechanism](https://kubernetes.io/docs/admin/daemons/)
-(to ensure auto-management of the pods). Agents come in two flavors - and there
- exist two daemonsets for each kind. The difference between them is that
-"Agent-hostnet" is tapped into host network namespace via supplying
-`hostNetwork: True` key-value for corresponding Pod's spec. As shown on diagram
-both daemonsets are enabled for each node meaning one and only one pod of each
-kind will be present there; consequently those pods
-uniquely represent nodes inside server's internals.
-
-The agents then periodically gather network related information
-(e.g. interfaces' info, results of nslookup, etc.) and send formed payload to
-the server address in the K8S cluster's network space.
-
-The server is deployed in dedicated Pod and exposed inside of the cluster via
-K8S service resource. Thus every agent can access the server by the service's
-DNS name.
-
-Incoming agent data is stored in server's memory (this fact then implicates
-loss of the data when server application is shutdown or restarted; it is going
-to be reworked by adding a permanent storage in the future).
-The cache is represented as Go map data structure with an agent's pod name as
-unique key.
-
-Server provides HTTP RESTful interface which for the time present consists of
-the following (verb - URI designator - meaning of the operation):
+are deployed on every Kubernetes node using
+[Daemonset](https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/).
+Agents come in two flavors - and default setup includes two corresponding
+daemonsets. The difference between them is that "Agent-hostnet" is tapped into
+host network namespace via supplying `hostNetwork: True` key-value for the
+corresponding Pod's specification. As shown on the diagram, both daemonsets
+are enabled for each node meaning exactly one pod of each kind will be deployed
+on each node.
+
+The agents periodically gather network related information
+(e.g. interfaces' info, results of nslookup, results of latencies measurement,
+etc.) and send it to the server as periodic agent reports.
+Report includes agent pod name and its node name so that the report is uniquely
+identified using them.
+
+The server is deployed in a dedicated pod using
+[Deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/)
+and exposed inside of the cluster via Kubernetes service resource. Thus, every
+agent can access the server by the service's DNS name.
+
+Server processes the incoming agent data (agents' reports) and store it in
+persistent data storage - using Kubernetes third party resources (TPR). New data
+type called `agent` was added into TPR, Kubernetes API was extended with this
```

Comment:
My meaning here was that "new data type called`agent`" is already a custom TPR. I mean it "is" not "as".